app/controllers/sitemaps_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SitemapsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[show]
+
+  def show
+    date_string = params[:sitemap].gsub("sitemap-", "").gsub(".xml", "")
+    not_found unless date_string.match?(/\A[A-Z][a-z][a-z]\-\d{4}\z/)
+    begin
+      date = Time.zone.parse(date_string).at_beginning_of_month
+    rescue ArgumentError
+      not_found
+    end
+    end_date = (date + 1.month).at_beginning_of_month
+    @articles = Article.published.where("published_at > ? AND published_at < ? AND score > ?", date, end_date, 3).pluck(:path, :last_comment_at)
+
+    response.headers["Surrogate-Control"] = if date > 1.month.ago
```

Comment:
Newer sitemaps expire sooner, all are "stale while re-validate" to some extent to ensure the bots don't have to fetch from cold.

Reply:
Maybe something like this, it's more compact and makes it a bit more obvious what changes between the two:

```ruby
max_age = date > 1.month.ago ? 8640 : 259200
response.headers["Surrogate-Control"] = "max-age=#{max_age}, stale-while-revalidate=43200, stale-if-error=86400"
```